src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,187 @@
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$first") && param.Schema.Type.Equals("integer")));
+                    Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name.Equals("$after") && param.Schema.Type.Equals("string")));
+                }
+                else
+                {
+                    // No query parameters for other OperationTypes. 
+                    Assert.IsFalse(operation.Parameters.Any(param => param.In is ParameterLocation.Query));
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Validates that Input parameters are generated for Stored Procedures with GET operation.
+    /// Input parameters for stored procedures will be generated for GET operation.
+    /// If a parameter default value is not provided in the config, it will marked as required.
+    /// It also validates parameter metadatas like type, name, location, required, etc.
+    /// </summary>
+    /// <param name="entityName">The name of the entity.</param>
+    /// <param name="objectName">The name of the database object.</param>
+    [DataTestMethod]
+    [DataRow("CountBooks", "count_books", DisplayName = "StoredProcudure with no parameters results in 0 created input query params.")]
+    [DataRow("UpdateBookTitle", "update_book_title", DisplayName = "StoredProcedure with parameters results in created input query params.")]
+    public async Task ValidateInputParametersForStoredProcedures(string entityName, string objectName)
+    {
+        Dictionary<string, object> parameterDefaults = null;
+        if (entityName.Equals("UpdateBookTitle"))
+        {
+            // Adding a parameter default value.
+            parameterDefaults = new Dictionary<string, object> { { "title", "Test Title" } };
+        }
+
+        EntitySource entitySource = new(Object: objectName, EntitySourceType.StoredProcedure, parameterDefaults, null);
+        OpenApiDocument openApiDocument = await GenerateOpenApiDocumentForGivenEntityAsync(entityName, entitySource);
+
+        OpenApiPathItem pathItem = openApiDocument.Paths.First().Value;
+        foreach ((OperationType operationType, OpenApiOperation operation) in pathItem.Operations)
+        {
+            if (entityName.Equals("UpdateBookTitle"))
+            {
+                if (operationType is OperationType.Get)
+                {
+                    Assert.IsTrue(operation.Parameters.Any(param =>
+                    param.In is ParameterLocation.Query
+                    && param.Name.Equals("id")
+                    && param.Schema.Type.Equals("number")
+                    && param.Required is true
+                    ));
+
+                    // Parameter with default value will be an optional query parameter.
+                    Assert.IsTrue(operation.Parameters.Any(param =>
```

Comment:
`IEnumerable.Single()` will ensure that the expected param is present only once and throw if that expectation fails.